Code warnings in bacio.c

```
cd /home/ed/NCEPLIBS-bacio/b/src && /usr/bin/cc -DLINUX -DUNDERSCORE  -g -g -fprofile-abs-path -fprofile-arcs -ftest-coverage -O0 -fsanitize=address -Wall -MD -MT src/CMakeFiles/bacio_4_c.dir/bacio.c.o -MF CMakeFiles/bacio_4_c.dir/bacio.c.o.d -o CMakeFiles/bacio_4_c.dir/bacio.c.o -c /home/ed/NCEPLIBS-bacio/src/bacio.c
/home/ed/NCEPLIBS-bacio/src/bacio.c: In function ‘bacio_’:
/home/ed/NCEPLIBS-bacio/src/bacio.c:165:7: warning: unused variable ‘tcharval’ [-Wunused-variable]
  165 |   int tcharval;
      |       ^~~~~~~~
/home/ed/NCEPLIBS-bacio/src/bacio.c:163:10: warning: unused variable ‘j’ [-Wunused-variable]
  163 |   int i, j, jret, seekret;
      |          ^
/home/ed/NCEPLIBS-bacio/src/bacio.c: In function ‘banio_’:
/home/ed/NCEPLIBS-bacio/src/bacio.c:518:10: warning: unused variable ‘count’ [-Wunused-variable]
  518 |   size_t count;
      |          ^~~~~
/home/ed/NCEPLIBS-bacio/src/bacio.c:517:7: warning: unused variable ‘tcharval’ [-Wunused-variable]
  517 |   int tcharval;
      |       ^~~~~~~~
/home/ed/NCEPLIBS-bacio/src/bacio.c:515:10: warning: unused variable ‘j’ [-Wunused-variable]
  515 |   int i, j, jret, seekret;
      |          ^
/home/ed/NCEPLIBS-bacio/src/bacio.c: In function ‘baciol_’:
/home/ed/NCEPLIBS-bacio/src/bacio.c:850:7: warning: unused variable ‘tcharval’ [-Wunused-variable]
  850 |   int tcharval;
      |       ^~~~~~~~
/home/ed/NCEPLIBS-bacio/src/bacio.c:848:10: warning: unused variable ‘j’ [-Wunused-variable]
  848 |   int i, j, jret, seekret;
      |          ^
/home/ed/NCEPLIBS-bacio/src/bacio.c: In function ‘baniol_’:
/home/ed/NCEPLIBS-bacio/src/bacio.c:1205:10: warning: unused variable ‘count’ [-Wunused-variable]
 1205 |   size_t count;
      |          ^~~~~
/home/ed/NCEPLIBS-bacio/src/bacio.c:1204:7: warning: unused variable ‘tcharval’ [-Wunused-variable]
 1204 |   int tcharval;
      |       ^~~~~~~~
/home/ed/NCEPLIBS-bacio/src/bacio.c:1202:10: warning: unused variable ‘j’ [-Wunused-variable]
 1202 |   int i, j, jret, seekret;
      |          ^
```

--- a/src/bacio.c
+++ b/src/bacio.c
@@ -160,9 +160,8 @@
           char *datary,
           long long int  namelen, long long int  datanamelen) {
 #endif
-  int i, j, jret, seekret;
+  int i, jret, seekret;
   char *realname;
-  int tcharval;
   size_t count;
 
 /* Initialization(s) */
@@ -515,9 +514,8 @@
           char *datary,
           long long int  namelen, long long int  datanamelen) {
 #endif
-  int i, j, jret, seekret;
+  int i, jret, seekret;
   char *realname;
-  int tcharval;
   size_t count;
 
 /* Initialization(s) */

